xmppserver/src/main/java/org/jivesoftware/openfire/net/TLSWrapper.java

```diff
@@ -136,10 +139,21 @@ public void enableLogging(boolean logging) {
     public ByteBuffer unwrap(ByteBuffer net, ByteBuffer app) throws SSLException {
         ByteBuffer out = app;
         out = resizeApplicationBuffer(out);// guarantees enough room for unwrap
+
+        // Record a hex dump of the buffer, but only when logging on level 'debug'.
+        String hexDump = null;
+        if ( Log.isDebugEnabled() )
```

Comment:
If debug is enabled, hexDump is calculated - but only ever used if the exception occurs. Is it worth moving this section to the exception handler?

Reply:
The intend is to capture the buffer before it gets touched (possibly modified) by the TLS handling.

Reply:
A comment to that effect would probably help prevent this being refactored in error in the future ...